proto/greenfield/virtualgroup/events.proto

```diff
@@ -165,12 +165,54 @@ message EventStorageProviderExit {
 message EventCompleteStorageProviderExit {
   // The id of the storage provider who complete exit
   uint32 storage_provider_id = 1;
-  // The operator address which initial the complete exit transaction
+  // The operator address which initials the complete exit transaction
   string operator_address = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // The storage address which completes exit transaction
+  string storage_provider_address = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
   // total_deposit defines the number of tokens deposited by this storage provider for staking.
-  string total_deposit = 3 [
+  string total_deposit = 4 [
     (cosmos_proto.scalar) = "cosmos.Int",
     (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
     (gogoproto.nullable) = false
   ];
+  //  forced_exit whether the exit is a forced exit
+  bool forced_exit = 5;
+}
+
+message EventSwapIn {
+  // The id of the storage provider who wants to swap in
+  uint32 storage_provider_id = 1;
+  // The id of the gvg family which the storage provider wants to swap in as primary sp
+  uint32 global_virtual_group_family_id = 2;
+  // The id of the gvg which the storage provider wants to swap in as secondary sp
+  uint32 global_virtual_group_id = 3;
```

Comment:
If there are many GVGs (usually it is the case), the successor needs to submit many transactions. How about supporting batch GVGs?

Reply:
the offchain process takes time, try to avoid a SP reserve too many GVG. It is recommended to reserve one and complete the data transfer before starting next.